docs/INSTALL.md

```diff
@@ -0,0 +1,214 @@
+5. After creating the user, click on "Edit", then navigate to the Permissions section and click on "Go to permissions".
+6. Enable all permissions except "Control Panel" and save the changes.
+
+
+
+### Creating Your First WHMCS Server
+
+Reference: https://docs.whmcs.com/Servers
+
+1. Log in to your WHMCS as an admin user and go to Configuration > System Settings > Servers.
+2. Select "Add New Server" and choose "UpCloud VPS" from the module dropdown.
+3. Enter the Hostname or IP Address as: api.upcloud.com (you can assign any name).
+4. For Username, input your API username created in the previous step.
+5. For Password, input your API password created in the previous step.
+6. Proceed by clicking on "Test Connection" or "Continue Anyway".
+
+### Creating a Server Group
+
+1. Click Create New Group under the Options heading.
+2. Enter a name for your group.
+3. Select the servers you want to assign (previously created) to this group in the box on the left.
+4. Click Add to move them to the box on the right, which contains the servers for this group.
+5. Click Save Changes to complete the process.
+
+## Configuration
+
+### Creating a Product Group
+
+1. Navigate to Configuration > System Settings > Product/Services
+
+![Producs & Services](images/products_services.png)
+
+
+2. Click on Create Product Group
+
+![Product Group](images/product_group.png)
+
+
+3. Provide a Product Group name and adjust all other settings based on your requirements
+
+![Product Config](images/product_config.png)
+
+4. Click on Save Changes
+
+
+### Creating a New Product
+
+1. Click on Create a New Product
+
+![New Product](images/new_product.png)
+2. Configure the following details:
+   - **Product Type**: Select as "Other" (recommended to avoid prompting customers for Hostname, Password, NS1 & NS2 prefix during order process)
+   - **Product Group**: Select the product group that you created
+   - **Product Name**: Provide your product name
+   - **Module**: UpCloud VPS
+   - **Create as Hidden**: Off
+3. Click on Save Changes
+
+### Module Settings
+
+On the next page:
+1. Verify the Product Type
+2. Set Welcome Email to "Dedicated/VPS Server Welcome Email"
+3. Go to the Module Settings Tab
+4. Select the required details:
+   - Module Name
+   - Server group
+   - Default Location
+   - Plan
+   - Template
+5. If for Plans you selected "Custom", then users will have the ability to choose their own CPU, RAM, and Storage space based on which VMs will be created.
+
+![Server Plan Config](images/server_plan_config.png)
+
+6. Decide whether the product requires manual approval from admins or is automatically deployed upon order.
+7. Click the Save Changes button.
+
+![Server Plan Ready](images/server_plan_ready.png)
+
+This will create custom fields named (`instanceId`, `ssh_key` & `userData`) as shown below:
+
+![Custom Fields](images/custom_fields.png)
+
+### Configurable Options
```

Comment:
Configurable option sets were recently changed in #68 , an update to this whole section seems to be in order.

Reply:
Updated to match the current implementation